s3transfer/download.py

```diff
@@ -153,6 +178,55 @@ def get_final_io_task(self):
             transfer_coordinator=self._transfer_coordinator)
 
 
+
+class DownloadNonSeekableOutputManager(DownloadOutputManager):
+    def __init__(self, osutil, transfer_coordinator, io_executor,
+                 defer_queue=None):
+        super(DownloadNonSeekableOutputManager, self).__init__(
+            osutil, transfer_coordinator, io_executor)
+        if defer_queue is None:
+            defer_queue = DeferQueue()
+        self._defer_queue = defer_queue
+        self._io_submit_lock = threading.Lock()
+
+    @classmethod
+    def is_compatible(cls, download_target):
+        # We're lying a bit here.  We actually *are* compatible
+        # with seekable fileobjs, we'll still queue the IO writes
+        # in order.  Technically works, but this is not ideal behavior when you
+        # could just use DownloadSeekableOutputManager.
+        # Perhaps this method makes more sense as "should_use()"
+        return (
+            not seekable(download_target) and
+            hasattr(download_target, 'write')
```

Comment:
Should we add a writeable() back compat that will check:
```py
if hasattr(download_target, 'writeable'):
    return download_target.writeable()
return hasattr(download_target, 'write)
```
Jordon is doing the same thing for nonseekable uploads but for ``readable``